### PR TITLE
Issue #119 - remove outgoing commits in between processing libraries

### PR DESCRIPTION
**Pull request by metalcanine**
Going to test this in holly to make sure it both doesn't blow up and that `hg strip` is available before requesting review

### CODE REVIEW

**Comment by tomrittervg** [comment]
We should put a bug on file to add a test for this; would would be non-trivial right now because when we run our tests we don't actually mach vendor anything...

**Comment by metalcanine** [comment]
so it turns out we don't have strip available, so I need to either find a way to make it available or come up with a different plan

**Comment by tomrittervg** [comment]
Try enabling it here https://searchfox.org/mozilla-central/source/taskcluster/docker/updatebot/hgrc

File: automation.py [hunk 1]
```diff
@@ -173,7 +176,11 @@ def _process_library(self, library):
             self.logger.log("%s is a brand new revision to updatebot." % (new_version), level=LogLevel.Info)
             self._process_new_job(library, new_version, timestamp)
 
-        # TODO #119: We need to remove any commits we made so the repo is clean for the next library.
+        # remove commits generated from processing this library, will return success
+        # regardless of if outgoing commits exist or not.
+        self.logger.log("Removing any outgoing commits before moving on.")
```
Comment (tomrittervg) [inline automation.py:181]: This needs `, level=LogLevel.Info`

**Review by tomrittervg — CHANGES_REQUESTED** [review]
This is going to need rebasing now that this code has moved to tasks/vendoring

**Review by tomrittervg — APPROVED** [review]

**Pull request merged**